include/swift/AST/DiagnosticsSema.def

```diff
@@ -2595,6 +2595,8 @@ WARNING(expression_unused_optional_try,none,
         "result of 'try?' is unused", ())
 WARNING(expression_unused_selector_result, none,
         "result of '#selector' is unused", ())
+WARNING(expression_unused_literal,none,
+        "%0 literal is unused", (StringRef))
```

Comment:
Yeah, I didn't see a great value that I could pass in for the select. But maybe I can out the literal descriptions in ExprNodes.def too.

How will folks remember to add their hypothetical new LITERAL_EXPR in ExprNodes?